innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/utils.py

```diff
@@ -535,6 +535,73 @@ def add_n_months_before_first_investment_in_window(
     return cb_data
 
 
+def add_n_unique_investors_total(
+    cb_data: pd.DataFrame,
+    cb_funding_rounds: pd.DataFrame,
+    cb_investments: pd.DataFrame,
+    start_date: pd.DatetimeIndex,
+    end_date: pd.DatetimeIndex,
+) -> pd.DataFrame:
+    """Add column for number of unique investors total
+
+    Args:
+        cb_data: Dataframe to add column for number of unique investors total to
+        cb_funding_rounds: Dataframe containing information about crunchbase
+            funding rounds
+        cb_investments: Dataframe containing information about crunchbase investments
+        start_date: Start date of the time window
+        end_date: End date of the time window
+
+    Returns:
+        cb_data with additional column for number of unique investors total
+
+    """
+    funding_rounds_in_window = (
+        cb_funding_rounds.astype({"announced_on": "datetime64[ns]"})
+        .query(f"'{start_date}' <= announced_on <= '{end_date}'")[["id", "org_id"]]
```

Comment:
Just a question to see if I understand: I guess you're anyways filtering out all companies that were founded before the `start_date` ? (so they couldn't have had any investments before the `start_date`).

Also (and I'm not 100% sure about this): Perhaps it might not be strictly necessary to consider only investors that have joined within the time window - and instead you could use the total number of all investors that have invested up until the end of the window? I guess this doesn't matter in our case as the window is quite wide, and you're anyways filtering the companies by founding date. But maybe it's something to keep in mind if we would consider having one window for including past investments, and another window for filtering companies by age...

Anyhow, I think it's fine now to keep this as it is!

Reply:
Yes, we are filtering out all companies that were founded before `start_date` and so there shouldn't be any investments before `start_date`.